### ISSUE TITLE
Download URLs for C GSW library and possible registration

### ISSUE THREAD

**Issue by Alexander-Barth** [ax1ine/GSW-Julia.jl#7]
Thank for your effort for making this package available. We are looking if we can depend on your implementation for our package `PhysOcean.jl`. It is nice to see that you build the library for so many platforms. I am wondering if there is an issue in the URL of the file https://github.com/ax1ine/GSW.jl/blob/master/deps/build.jl which results in a problem during the installation:

```
 Building GSW → `~/.julia/dev/GSW/deps/build.log`
┌ Error: Error building `GSW`: 
│ [ Info: Downloading https://github.com/ax1ine/GSW/releases/download/v0.1.0/GSW_BinaryBuilder.v0.1.0.x86_64-linux-gnu.tar.gz to /home/abarth/.julia/dev/GSW/deps/usr/downloads/GSW_BinaryBuilder.v0.1.0.x86_64-linux-gnu.tar.gz...
│ ERROR: LoadError: Could not download https://github.com/ax1ine/GSW/releases/download/v0.1.0/GSW_BinaryBuilder.v0.1.0.x86_64-linux-gnu.tar.gz to /home/abarth/.julia/dev/GSW/deps/usr/downloads/GSW_BinaryBuilder.v0.1.0.x86_64-linux-gnu.tar.gz:
│ ErrorException("")
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
...
```

Did you also consider to register your package so that other package can depend on your implementation?

Thanks again for your effort!

**Comment by kouketsu** [comment]
With [@Alexander-Barth comment](https://github.com/kouketsu/GSW.jl/issues/1), I have improved [my package](https://github.com/kouketsu/GSW.jl).

But I think @ax1ine work here is better because GSW.jl with clear comments is useful for other users. So I would like to recommend to consider registering your package, too.

If you consider providing a binary for OSX environment before registering, you can check my repository (which were made following @Alexander-Barth comment).

**Comment by Alexander-Barth** [comment]
If you are interested in a common home for the GSW.jl package, then the github organization [JuliaGeo](https://github.com/JuliaGeo/) might be such a place.

**Comment by ax1ine** [comment]
Hi @Alexander-Barth and @kouketsu ! I think it's a great idea to join forces and put GSW.jl at JuliaGeo.  I haven't been much active with developing of my package recently (another project, other priorities, as usual) but I'd be happy to contribute. Btw, there's one more package for TEOS-10 called [GibbsSeaWater.jl](https://github.com/njwilson23/GibbsSeaWater.jl). The author is responsive and it might be useful to have him on board, as well.
@kouketsu , would you be ok with registering your package as the main one? I could port comments from my package if those are useful. Your binaries are newer and support more platforms so I'd stick with using them.

**Comment by Alexander-Barth** [comment]
@meggart, @evetion, @yeesian, is there a process for getting new members to the JuliaGeo organization? This GSW library computes basic characteristic of sea water (density, temperature, conductivity, pressure,...) and is the reference library for physical oceanographers. There have been at least 3 different implementations but none reached the state to be registered. @kouketsu and @ax1ine are interested to join forces to work on a common package. I would also be interested to contribute to it. Can I simply invite them to become member, or should there be some form of approval necessary?  

Maybe it is safer to use a slightly longer name than GSW for the common repo. What do you think?
https://julialang.github.io/Pkg.jl/v1/creating-packages/index.html#Package-naming-guidelines-1

**Comment by kouketsu** [comment]
@ax1ine I'm OK with registering mine.
I think following the guideline is good.

I think we also had better inform TEOS-10 team (https://github.com/TEOS-10), where the packages for various languages were listed up.

**Comment by yeesian** [comment]
> is there a process for getting new members to the JuliaGeo organization?

Hi @Alexander-Barth, the process is described here: https://github.com/JuliaGeo/meta/wiki/How-to-get-Involved

The JuliaGeo organization is loosely-tied (in name and focus) to https://www.osgeo.org/. If there is a collection of packages that works well together for oceanography (that doesn't otherwise depend on JuliaGeo), it might make sense to create a separate github organization for it -- that might give you more freedom to take it in a direction that facilitates development for those packages you are interested in.

**Comment by Alexander-Barth** [comment]
Dear @yeesian, the GSW packages does not have any dependencies (to JuliaGeo or otherwise). Should the "Geo" in JuliaGeo be interpreted as "Geo" as in Geography/Geodesy or as in Geosciences (including also Oceanography, Meteorology, ...). From you message, it seem that rather the former is true. Can you confirm?

**Comment by kouketsu** [comment]
Thanks @yeesian.
@ax1ine and @Alexander-Barth, I think we should prepare to register GSW as an official package, at first, which can make users to install the package easily. I think that is you need @Alexander-Barth.
After that, we consider registering this to JuliaGeo and TEOS-10  communities. 

To follow the guidelines for official packages, I would like to change the package name to "GibbsSeaWater". As the name has been used by @njwilson23, I will contact with @njwilson23.

OK?

But sorry, I will do these slowly because I will be on board from the next week.

**Comment by yeesian** [comment]
> Should the "Geo" in JuliaGeo be interpreted as "Geo" as in Geography/Geodesy or as in Geosciences (including also Oceanography, Meteorology, ...). From you message, it seem that rather the former is true. Can you confirm?

Yeah, I think that is true. Membership in an organization/team are a way of managing permission settings for packages that you co-develop with people, and are best used for that purpose. I'm sorry about the ambiguity in the name, here is the thread where the discussion for it happened: https://groups.google.com/d/msg/julia-geo/oPz2_gz9K-g/TbULPveFXdAJ.

**Comment by njwilson23** [comment]
> To follow the guidelines for official packages, I would like to change the package name to "GibbsSeaWater". As the name has been used by @njwilson23, I will contact with @njwilson23.

I'd be happy to have the name be used by an actively maintained GSW package.

**Comment by Alexander-Barth** [comment]
@yeesian thanks you very much for the clarification. After all, it is the original meaning of the name "Geo" :-)  Should we contact https://github.com/TEOS-10 to see if they want to host the julia package? It seems that they have several repos for different languages (some empty like the one for IDL or octave, probably placeholders for potential contributors).

**Comment by ax1ine** [comment]
@Alexander-Barth , I think it's a good idea to contact the TEOS-10 team. I found a bug in GSW a few year ago and they were responsive and helpful with fixing it. Who should do that? Would it be useful to draw their attention to BinaryBuilder.jl, so they could provide binaries themselves? AFAIK, their Python wrapper needs gcc to compile the code during installation. I never managed to get it work on Windows due to that. So there's a slim chance that they might be interested in providing precompiled binaries.

**Comment by kouketsu** [comment]
@ax1ine and @Alexander-Barth, my colleague is one of the maintainers of the TEOS-10 wrappers. So I can contact the team through him.

And, as far as I remember, there were some discussions to rearrange code to correspond to the windows environment (by E. Firing who is interested in Python wrapper, and some researchers).

**Comment by kouketsu** [comment]
Thanks!!, @njwilson23.

> > To follow the guidelines for official packages, I would like to change the package name to "GibbsSeaWater". As the name has been used by @njwilson23, I will contact with @njwilson23.
> 
> I'd be happy to have the name be used by an actively maintained GSW package.

**Comment by ax1ine** [comment]
@kouketsu , that would be great, thank you!
> my colleague is one of the maintainers of the TEOS-10 wrappers. So I can contact the team through him.
>

@njwilson23 , Thank you, Nat! I didn't find the time to finalize my changes into a stable working package. Hope, as a community we'll be able to release it and keep it maintained.
> I'd be happy to have the name be used by an actively maintained GSW package.

**Comment by Alexander-Barth** [comment]
For your information, I contributed the GSW build script also to Yggdrasil:

https://github.com/JuliaPackaging/Yggdrasil/pull/216

**Comment by PaulMBarker** [comment]
Hi,
I have made the repository at TEOS-10 on github. and have invited kouketsu. Please let me know if you would like an invite.
Paul.

**Comment by Alexander-Barth** [comment]
Thanks Paul! I would be interested to contribute to too. So far I am using the old EOS-80 equations in https://github.com/gher-ulg/PhysOcean.jl but I would be interested to use support also TEOS 10.

**Comment by ax1ine** [comment]
@PaulMBarker , thank you very much! Please, invite me, as well. 
I also wrote a FreePascal/Lazarus (it may work with Delphi, too) wrapper for the 'C' version of TEOS-10. Would it be useful to add it to the main TEOS-10 repository?

**Comment by PaulMBarker** [comment]
yes please do add your FreePascal/Lazarus wrapper to the github site. I will also add this to the main TEOS-10  page. I have a c++ wrapper to add too. 
Let me know if you need any more permissions to add this.

**Comment by ax1ine** [comment]
@PaulMBarker , Paul, could you, please, create a repository called 'GSW_FreePascal', and invite me there? I'll add the wrapper and some text on how to work with it.

**Comment by kouketsu** [comment]
@PaulMBarker, thanks.
@ax1ine and @Alexander-Barth, I have uploaded my repository to the TEOS-10 official site. I will be happy to receive any requests on the official repository (while I will be on board by 2020/2 and my response will be very slow ;)).

**Comment by ax1ine** [comment]
@kouketsu great, thank you! Do you mind if I add descriptions to the functions (like I did in my old module)?

@PaulMBarker Thank you for the repository, Paul! I'll upload the FreePascal code this weekend (performing some last tests now).

**Comment by kouketsu** [comment]
@ax1ine, I never mind and will appreciate your contribution :).
